### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by ReneLombard**
This pull request introduces several improvements and new features across the codebase, focusing on enhanced error handling, improved test coverage, new Cucumber BDD test suites, and CI workflow updates. The most significant changes are the addition of Cucumber feature files and configuration for smoke and integration testing, stricter error handling in the code generator, and more robust and explicit test cases for edge scenarios.

**Testing and Feature Coverage Improvements:**

* Added Cucumber BDD configuration (`src/cucumber.js`) and multiple feature files (`src/features/00-smoke-tests.feature`, `src/features/01-project-initialization.feature`, `src/features/02-mermaid-transformation.feature`) to provide smoke, initialization, and transformation scenario coverage, supporting both core and edge-case functionality. [[1]](diffhunk://#diff-314bfa186f45350a326123cc9fae763fa6aeae1b7c36f8318dd2563f99c54584R1-R23) [[2]](diffhunk://#diff-59bd69f747e9e069b5c9a0ba957acb3751b229bcfe4e7275c4276ade98ce5f2bR1-R37) [[3]](diffhunk://#diff-d7856d5d1a0cf54160756089e3e6c1c038507a81f8911bd8303245046142d454R1-R28) [[4]](diffhunk://#diff-688b400b16f413bea30856670e3a34e9f60f817b706e7403448fbff98f9820a4R1-R64)

* Updated and expanded Jest test cases for `CodeGenerator`, `InitializeService`, and `ListLanguagesService` to use more precise error assertions, mock behaviors, and improved coverage for directory and template existence checks. [[1]](diffhunk://#diff-27af44c72847805febddeffd40c5f80d9feda39b68f4b375e110b34f5cbc6c78L50-R76) [[2]](diffhunk://#diff-535e1492a6f2807a7acba94fbe4ba1a09b73d35e61795de6a8242da885e1869fL69-R75) [[3]](diffhunk://#diff-535e1492a6f2807a7acba94fbe4ba1a09b73d35e61795de6a8242da885e1869fL90-R100) [[4]](diffhunk://#diff-535e1492a6f2807a7acba94fbe4ba1a09b73d35e61795de6a8242da885e1869fL105-R120) [[5]](diffhunk://#diff-0cabedeb4f92b189f11106710a16f33ba2c0cb10ac4e2c67a781042bbf06ec39R28) [[6]](diffhunk://#diff-0cabedeb4f92b189f11106710a16f33ba2c0cb10ac4e2c67a781042bbf06ec39R50) [[7]](diffhunk://#diff-0cabedeb4f92b189f11106710a16f33ba2c0cb10ac4e2c67a781042bbf06ec39L57-R70) [[8]](diffhunk://#diff-0cabedeb4f92b189f11106710a16f33ba2c0cb10ac4e2c67a781042bbf06ec39R96) [[9]](diffhunk://#diff-0cabedeb4f92b189f11106710a16f33ba2c0cb10ac4e2c67a781042bbf06ec39R115-R131)

**Error Handling Enhancements:**

* Refactored `CodeGenerator` to throw explicit errors (instead of logging and returning) when input, template, or output directories do not exist, resulting in clearer error propagation and simpler test assertions.

**Functionality and Helper Improvements:**

* Added a `capitalize` helper to Handlebars registration in `CodeGenerator` to support template capitalization needs.

**Continuous Integration Workflow Updates:**

* Updated the GitHub Actions workflow (`.github/workflows/npm-publish-packages.yml`) to:
  - Run build and tests on both push and pull request events to `main`
  - Use a unified test command for unit and integration tests
  - Restrict version bump and docs deployment jobs to run only on `main` branch push events (not on PRs) [[1]](diffhunk://#diff-ff6cb16b1d28edc054ccfa718ff08715379ac358d16368be0b56ae01a230ee54L1-R9) [[2]](diffhunk://#diff-ff6cb16b1d28edc054ccfa718ff08715379ac358d16368be0b56ae01a230ee54L23-R34) [[3]](diffhunk://#diff-ff6cb16b1d28edc054ccfa718ff08715379ac358d16368be0b56ae01a230ee54R60)

**Test Robustness and Accuracy:**

* Updated integration tests for Mermaid transformation to expect errors on invalid or empty diagrams, ensuring that parsing failures are surfaced and handled as expected. [[1]](diffhunk://#diff-a311f5b71259c091b33917f15fea118cc307f257a6ebea4db9ff8a966e1efa64L241-R242) [[2]](diffhunk://#diff-a311f5b71259c091b33917f15fea118cc307f257a6ebea4db9ff8a966e1efa64L258-R260) [[3]](diffhunk://#diff-a311f5b71259c091b33917f15fea118cc307f257a6ebea4db9ff8a966e1efa64L400-R403)

### CODE REVIEW

File: src/features/01-project-initialization.feature [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  Scenario: Initialize a C# project
+    Given I have selected "C#" as the target language
+    And I have specified "test-project" as the project directory
+    When I run the initialize command
+    Then a new project structure should be created
+    And the project should contain C# specific templates
+    And the command should return exit code 0
+
+  Scenario: List available languages
+    When I run the list languages command
```
[the remark this answers is not in the document]
Comment (ReneLombard) [reply src/features/01-project-initialization.feature]: Reworked

File: .github/workflows/copliot-instructions.feature.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+- **When**: Actions/events being tested (present tense)
+    - Example: `When Bob enters a username that consists of 2 characters`
+- **Then**: Expected outcomes/results (present tense)
+    - Example: `Then the software does not accept the username`
+- **And**: Continue previous step type (indented one level more)
+    - Example: `And Bob started the application`
+
+### Writing Best Practices
+
+#### Make Steps "Alive"
```
Comment (djvandijk) [inline .github/workflows/copliot-instructions.feature.md]: I know where this suggestion is coming from and some people actually liked it, but in this case of the mermaid codegenerator I don't think it adds much. There aren't multiple user roles or multiple personas using this tool. It's not that there are, for instance, different persons performing the transformations and others are doing the code generating. 
Not sure how this works out in vscode, but in visual studio with the specflow plugin this actually made it hard to find the steps you are looking for since all steps start with a variable. The code completion there doesn't like it.

Comment (ReneLombard) [reply .github/workflows/copliot-instructions.feature.md]: Agreed. For the Mermaid code generator, this does not add much value. I’m also not aware that it’s possible to step into JavaScript code from feature files in Visual Studio or VS Code.

For the sake of completing this pull request, I would keep it as it is.

**Review by djvandijk — APPROVED** [review]
